plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.sdk;
+
+import com.google.cloud.tools.appengine.cloudsdk.CloudSdk;
+import com.google.cloud.tools.eclipse.sdk.internal.CloudSdkInstallJob;
+import com.google.cloud.tools.managedcloudsdk.ManagedCloudSdk;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVerificationException;
+import com.google.cloud.tools.managedcloudsdk.ManagedSdkVersionMismatchException;
+import com.google.cloud.tools.managedcloudsdk.UnsupportedOsException;
+import com.google.cloud.tools.managedcloudsdk.components.SdkComponent;
+import java.nio.file.Path;
+
+public class CloudSdkManager {
+
+  /**
+   * Returns {@code CloudSdk} only if there exists an up-to-date Cloud SDK with the App Engine Java
+   * component installed at the managed location. Otherwise, returns {@code null} after triggering a
+   * background installation task. Callers who get {@code null} may need to try this method at later
+   * points.
+   *
+   * Note that even if the Cloud SDK is already installed, this method executes {@code gcloud} twice
+   * externally to check components and versions, which may incur a noticeable delay. If possible,
+   * avoid unnecessary repetitive calls.
+   *
+   * @return {@code CloudSdk} if there exists an up-to-date Cloud SDK with the App Engine Java
+   *     component installed at the managed location; {@code null} otherwise
+   * @throws ManagedSdkVerificationException if unable to verify SDK installation; possible causes
+   *     include corrupted installation or a problem executing binaries
+   */
+  public static CloudSdk getCloudSdk() throws ManagedSdkVerificationException {
+    try {
+      ManagedCloudSdk managedSdk = ManagedCloudSdk.newManagedSdk();
+      if (!managedSdk.isInstalled()
+          || !managedSdk.hasComponent(SdkComponent.APP_ENGINE_JAVA)
+          || !managedSdk.isUpToDate()) {
```

Comment:
The isUpTodate check should probably be removed since that's the one that hits the network.

Reply:
I think it's a good idea. We could rather check updates periodically in background as you mentioned earlier.

Reply:
It turns out `hasComponent()` (effectively `gcloud components list`) hits the network too.

```
$ gcloud components list
ERROR: (gcloud.components.list) Failed to fetch component listing from server. Check your network settings and try again.
$ 
```